commons-db/src/main/java/com/palantir/nexus/db/sql/SqlConnectionHelper.java

```diff
@@ -688,22 +691,26 @@ void clearTempTableUsingDelete(Connection c, String tempTable) throws PalantirSq
             executeUnregisteredQuery(c, sqlDelete, new Object[]{}); // dynamic query
         } catch (PalantirSqlException e) {
             if (e.getMessage().contains("ORA-08102")) { // "index key not found"
-                // Try again
+                // Gather some information
+                AgnosticResultSet dbaRows = null;
+                String objectIdString = null;
                 try {
-                    executeUnregisteredQuery(c, sqlDelete, new Object[]{}); // dynamic query
-                    // Worked the second time
-                    // In production we can continue, but if asserts are on let's fail with an informative message to help diagnosis
-                    assert false: String.format("Tried to delete temp table %s, got ORA-08102, then tried again and it worked: %s",
-                            tempTable, sqlDelete);
-                } catch (PalantirSqlException e2) {
-                    // Repeated failure
-                    throw new RuntimeException(String.format(
-                            "Repeatedly failed to delete temp table %s: %s.  First exception was ORA-08102.  Second was: %s",
-                            tempTable, sqlDelete, e2.getMessage()), e2);
+                    Pattern p = Pattern.compile(".*ORA-08102: index key not found, obj# (\\d+),.*");
```

Comment:
would be nicer to not have to re-compile by putting this pattern outside of here